### DISCUSSION TITLE
How to prevent event bubbles in Nicegui?

### DISCUSSION THREAD

**Discussion by Autism-al** [zauberzeug/nicegui#3805] [Q&A] [answered]
### Question

I hope that clicking on an icon in a button only triggers the click event of that icon and not the click event of the button. Just like in the picture, when I clicked the delete icon, I hoped the buttons wouldn't be clicked together, and I didn't seem to find a solution in the document
![Uploading QQ图片20240929120140.png…]()

**Comment by python-and-novella** [comment]
Where is pic?

**Comment by Autism-al** [comment]
![QQ图片20240929120140](https://github.com/user-attachments/assets/b0c149ed-3199-4d4a-b66c-2cab3e9c8e49)

**Reply by python-and-novella** [reply]
Do you have a example you code?

**Reply by Autism-al** [reply]
I found that there was actually an error in my code, and I have just solved the problem. Sorry to bother you.

**Comment by python-and-novella** [comment, marked as answer]
```python3
with ui.button('Item').classes('w-96') as button:
    button.on_click(lambda :ui.notify('button'))
    ui.space()
    icon = ui.icon('delete').classes('z-0')
    icon.on('click',js_handler='(e)=>{e.stopPropagation()}')
    icon.on('click',lambda :ui.notify('icon'))
```

**Reply by Autism-al** [reply]
Thank you very much, this should be helpful for me in the future!
